main.tf

```diff
@@ -303,3 +305,175 @@ resource "aws_cloudfront_distribution" "cloudfront_distribution" {
     }
   }
 }
+
+resource "aws_cloudfront_distribution" "cloudfront_staging_distribution" {
+  count               = var.enable_cloudfront_staging ? 1 : 0
+  comment             = "${var.comment}. Staging associated with resource ID ${random_id.this.hex}"
+  default_root_object = var.default_root_object
+  enabled             = var.enable_cloudfront_staging
+  http_version        = "http2"
+  is_ipv6_enabled     = "true"
+  price_class         = var.cloudfront_price_class
+  retain_on_delete    = var.retain_on_delete
+  wait_for_deployment = var.wait_for_deployment
+  web_acl_id          = var.web_acl_id
+  staging             = true
+  dynamic "origin" {
+    for_each = var.dynamic_s3_origin_config_staging
+    content {
+      domain_name = origin.value.domain_name
+      origin_id   = origin.value.origin_id
+      origin_path = origin.value.origin_path
+      dynamic "s3_origin_config" {
+        for_each = origin.value.origin_access_identity != null ? [1] : []
+        content {
+          origin_access_identity = origin.value.origin_access_identity
+        }
+      }
+    }
+  }
+  dynamic "origin" {
+    for_each = var.dynamic_custom_origin_config_staging
+    content {
+      domain_name = origin.value.domain_name
+      origin_id   = origin.value.origin_id
+      origin_path = origin.value.origin_path
+      custom_origin_config {
+        http_port                = origin.value.http_port
+        https_port               = origin.value.https_port
+        origin_keepalive_timeout = origin.value.origin_keepalive_timeout
+        origin_read_timeout      = origin.value.origin_read_timeout
+        origin_protocol_policy   = origin.value.origin_protocol_policy
+        origin_ssl_protocols     = origin.value.origin_ssl_protocols
+      }
+      dynamic "custom_header" {
+        for_each = origin.value.custom_header
+        content {
+          name  = custom_header.value.name
+          value = custom_header.value.value
+        }
+      }
+    }
+  }
+  dynamic "origin_group" {
+    for_each = var.dynamic_origin_group_staging
+    content {
+      origin_id = origin_group.value.id
+      failover_criteria {
+        status_codes = origin_group.value.status_codes
+      }
+      member {
+        origin_id = origin_group.value.member1
+      }
+      member {
+        origin_id = origin_group.value.member2
+      }
+    }
+  }
+  default_cache_behavior {
+    allowed_methods         = var.default_cache_behavior.allowed_methods
+    cached_methods          = var.default_cache_behavior.cached_methods
+    target_origin_id        = var.default_cache_behavior.target_origin_id
+    compress                = var.default_cache_behavior.compress
+    viewer_protocol_policy  = var.default_cache_behavior.viewer_protocol_policy
+    realtime_log_config_arn = var.default_cache_behavior.realtime_log_config_arn
+
+    cache_policy_id          = aws_cloudfront_cache_policy.ordered_behaviors["default"].id
```

Comment:
WOuld it make sense to allow the possibility to specify a different cache behavior for the staging distribution? It's probably not in the migration scope, but it would allow to use continuos deployment also in other contexts